packages/webpack-cli/lib/utils/cli-flags.js

```diff
@@ -201,7 +201,7 @@ module.exports = {
             alias: 'p',
             usage: '--prod',
             type: Boolean,
-            defaultValue: undefined,
+            defaultValue: false,
```

Comment:
I've always wondered why we have flags `--dev`/`--prod`, we already have `--mode`, it can be misleading for new developers and some of them will be write `webpack --prod --mode=production`, but it is wrong

Reply:
> I've always wondered why we have flags --dev/--prod, we already have --mode, it can be misleading for new developers and some of them will be write webpack --prod --mode=production, but it is wrong

It's just alias for `--mode=development or production` like we have `--verbose` for `--stats=verbose`. 
CLI throws a warning if user do something like `webpack --prod --mode=production` 
`You provided both 'mode' and --dev arguments. You should provide just one. "mode" will be used`

Reply:
I think we should remove them in the future, but it is not high priority

Reply:
/cc @webpack/cli-team anyone in favor of `--dev` and `--prod`.? 
> Not a big deal but even if I understand the point of having clean and concise --dev and --prod options, I think the --mode= option better matches the way it works in configuration file and is more straightforward when you already know configuration file. Just my 2 cents, but I'm not sure a breaking change in this area is really useful ?

from feedback too https://github.com/webpack/webpack-cli/issues/1222#issuecomment-585678755